packages/driver/src/cy/assertions.ts

```diff
@@ -347,6 +349,8 @@ export const create = (Cypress, cy) => {
         if (_.isFunction(onRetry)) {
           return cy.retry(onRetry, options)
         }
+
+        return
```

Comment:
Do we need this trailing empty return? It looks very unusual to my eyes.

Reply:
It's because TypeScript wants all return paths. 

As the type of `cy.retry` became specific, TypeScript learned that this function returns a `Promise` above but nothing at the end.

Reply:
Without this it's an error.